### PR TITLE
MES-941: Prevent proceed beyond office screen before mandatory fields populated

### PR DESCRIPTION
**Pull request by johnmorleybjss**
* Ensure that route number, weather, show me question, independent driving and physical description are all completed.
* In the event of a fail due to dangerous/serious, ensure that each of those faults has an associated note.
* ~~Disable the submit button unless the above criteria are met.~~
* Give an alert when submit is pressed without the form being complete. Ensure the examiner _can_ defer completing the form if they have another imminent test.

### CODE REVIEW

File: src/components/post-test-summary-section/post-test-summary-section.ts [hunk 1]
```diff
@@ -37,4 +39,16 @@ export class PostTestSummarySectionComponent {
     textboxModal.onDidDismiss((notes?: string) => (this.faultNotes[faultName] = notes));
     textboxModal.present();
   }
+
+  isComplete(): boolean {
+    if (!this.canComplete) {
+      return true;
+    }
+    const notes = Object.keys(this.faultNotes)
```
Comment (ammarhaiderbjss) [inline src/components/post-test-summary-section/post-test-summary-section.ts]: Could use `Object.values` instead? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values

File: src/pages/post-test-summary/post-test-summary.ts [hunk 2]
```diff
@@ -72,7 +79,33 @@ export class PostTestSummaryPage {
...
+        .create({
+          title,
+          message,
+          buttons: [
+            {
+              text: 'Cancel'
+            },
+            {
+              text: 'Defer',
+              handler: () => this.backToJournal()
```
Comment (ammarhaiderbjss) [inline src/pages/post-test-summary/post-test-summary.ts]: Do we need this additional wrapper? can we just pass `this.backToJournal` instead?

File: src/components/post-test-summary-section/post-test-summary-section.ts (hunk 1 above)
Comment (johnmorleybjss) [reply src/components/post-test-summary-section/post-test-summary-section.ts]: Attempted to use this first off, but we don't have support, we'd need to configure TS for ES2017. Happy to make the change if that's what we want.

File: src/pages/post-test-summary/post-test-summary.ts (hunk 2 above)
Comment (johnmorleybjss) [reply src/pages/post-test-summary/post-test-summary.ts]: I tried without the wrapping arrow, I had a `this` binding issue.

File: src/components/post-test-summary-section/post-test-summary-section.ts (hunk 1 above)
Comment (ammarhaiderbjss) [reply src/components/post-test-summary-section/post-test-summary-section.ts]: ok nvm

File: src/pages/post-test-summary/post-test-summary.ts (hunk 2 above)
Comment (ammarhaiderbjss) [reply src/pages/post-test-summary/post-test-summary.ts]: fair enough, missed that this was all happening in a`.create`

File: src/components/post-test-summary-section/post-test-summary-section.ts (hunk 1 above)
Comment (johnmorleybjss) [reply src/components/post-test-summary-section/post-test-summary-section.ts]: Upgraded to ES2017, confirmed that `Object.values` is working as expected.

**Pull request merged**